docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,664 @@
+  - `--cluster_domain=` to the dns domain prefix to use for cluster DNS addresses.
+  - `--docker_root=`
+  - `--root_dir=`
+  - `--configure-cbr0=` (described above)
+  - `--register-node` (described in [Node](../node.md) documentation.
+
+#### Configuring kube-proxy
+
+All nodes should run kube-proxy.  (Running kube-proxy on a "master" node is not
+strictly required, but being consistent is easier.)   Obtain a binary as described for
+kubelet. 
+
+Arguments to consider:
+  - If following the HTTPS security approach:
+    - `--api_servers=https://$MASTER_IP`
+  - Otherwise, if taking the firewall-based security approach
+    - `--api_servers=http://$MASTER_IP`
+    - `--kubeconfig=/var/lib/kube-proxy/kubeconfig`
+
+#### Node networking setup
+Each node needs to be allocated its own CIDR range for pod networking.
+Call this $NODE_X_POD_CIDR.
+
+A bridge called `cbr0` needs to be created on each node.  The bridge is explained
+further in the [networking documentation](../networking.md). The recommended way to 
+do this is to let kubelet do it by passing `--configure-cbr0=true` to kubelet.
+Alternatively, the bridge can be created manually, like this:
+- Create a bridge
+  - e.g. `brctl addbr cbr0`.
+- Set appropriate MTU
+  - `ip link set dev cbr0 mtu 1460`
+- Add the clusters network to the bridge (docker will go on other side of bridge).
+  - e.g. `ip addr add $CLUSTER_CIDR dev eth0`
+**--configure-cbr0**=false
+	If true, kubelet will configure cbr0 based on Node.Spec.PodCIDR.
+- Turn it on
+  - e.g. `ip link set dev cbr0 up`
+- See `cluster/saltbase/salt/_states/container_bridge.py`
+
+If you have turned off docker ip masquerading to allow pods to talk to each
+other, then you may need to do masquerading just for destination IPs outside
+the cluster network.  For example:
+```iptables -w -t nat -A POSTROUTING -o eth0 -j MASQUERADE \! -d ${CONTAINER_SUBNET}```
+will rewrite the source address from
+the PodIP to the Node IP for traffic bound outside the cluster, and kernel
+[connection tracking](http://www.iptables.info/en/connection-state.html#UDPCONNECTIONS)
+will ensure that responses destined to the node still reach
+the pod.
+
+#### Other configuration
+- Configure auto-upgrades on node, if desired.
+- Configure log rotation for all node components (e.g. using [logrotate](http://linux.die.net/man/8/logrotate)).
+- Setup liveness-monitoring (e.g. using [monit](http://linux.die.net/man/1/monit)).
+- Setup volume plugin support (optional)
+  - Install any client binaries for optional volume types, such as `glusterfs-client` for GlusterFS
+    volumes.
+
+#### Using Configuration Management
+The previous steps all involved "conventional" system administration techniques for setting up
+machines.  You may want to use a Configuration Management system to automate the node configuration
+process.  There are examples of Saltstack, Ansible, Juju, and CoreOS Cloud Config in the
+various Getting Started Guides.
+
+### Bootstrapping the Cluster
+
```

Comment:
Can we insert a TOC (with links) of this section here?

 - Etcd
 - Apiserver
 - Scheduler
 - Controller Manager
 - Resources and Addons